### PR TITLE
[iOS] Allow returning multiple scripts per wallet provider

### PR DESCRIPTION
**Pull request by kylehickinson**
Also moves all of the wallet specific code out of `BraveCoreMain`

Resolves https://github.com/brave/brave-browser/issues/24781

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: ios/browser/api/brave_wallet/brave_wallet_api.mm [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+    return nil;
+  }
+
+  auto* brave_wallet_service =
+      brave_wallet::BraveWalletServiceFactory::GetServiceForState(browserState);
+  if (!brave_wallet_service) {
+    return nil;
+  }
+
+  auto* provider = new brave_wallet::EthereumProviderImpl(
```
Comment (Brandon-T) [inline ios/browser/api/brave_wallet/brave_wallet_api.mm:89]: This leaks.
The `brave_wallet::EthereumProviderImpl` takes a pointer, stores it as a `raw_ptr` but it never deletes it for you

File: ios/browser/api/brave_wallet/brave_wallet_api.mm [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+    return nil;
+  }
+
+  auto* tx_service =
+      brave_wallet::TxServiceFactory::GetServiceForState(browserState);
+  if (!tx_service) {
+    return nil;
+  }
+
+  auto* provider = new brave_wallet::SolanaProviderImpl(
```
Comment (Brandon-T) [inline ios/browser/api/brave_wallet/brave_wallet_api.mm:125]: This leaks.
The `brave_wallet::SolanaProviderImpl` takes a pointer, stores it as a raw_ptr but it never deletes it for you

**Review by Brandon-T — CHANGES_REQUESTED** [review]

File: ios/browser/api/brave_wallet/brave_wallet_api.mm (hunk 1 above)
Comment (kylehickinson) [reply ios/browser/api/brave_wallet/brave_wallet_api.mm]: Fixed in https://github.com/brave/brave-core/pull/14703/commits/880af8f9b696d95a7013ac4e8e37ced6089b1e4c

File: ios/browser/api/brave_wallet/brave_wallet_api.mm (hunk 2 above)
Comment (kylehickinson) [reply ios/browser/api/brave_wallet/brave_wallet_api.mm]: Fixed in https://github.com/brave/brave-core/pull/14703/commits/880af8f9b696d95a7013ac4e8e37ced6089b1e4c

**Review by Brandon-T — APPROVED** [review]

**Comment by kylehickinson** [comment]
If uplifted, also uplift https://github.com/brave/brave-core/pull/14722

**Pull request merged**